dbms/src/Processors/Sources/SourceFromInputStream.cpp

```diff
@@ -123,7 +123,9 @@ Chunk SourceFromInputStream::generate()
         return {};
     }
 
+#ifndef NDEBUG
```

Comment:
Should this be enabled for non-debug builds too already?

Reply:
I assume that this check may affect performance. It is mainly useful in tests to find out that something is broken. If we enable it in release build user will only get a strange error.